gslib/wildcard_iterator.py

```diff
@@ -697,12 +693,26 @@ def _IterDir(self, directory, wildcard):
     # originated on Windows) os.walk() will not attempt to decode and then die
     # with a "codec can't decode byte" error, and instead we can catch the error
     # at yield time and print a more informative error message.
-    for dirpath, dirnames, filenames in os.walk(six.ensure_text(directory)):
-      if self.logger:
-        for dirname in dirnames:
-          full_dir_path = os.path.join(dirpath, dirname)
-          if os.path.islink(full_dir_path):
-            self.logger.info('Skipping symlink directory "%s"', full_dir_path)
+    for dirpath, dirnames, filenames in os.walk(six.ensure_text(directory),
+                                                topdown=True):
+      filtered_dirnames = []
+
+      for dirname in dirnames:
+        full_dir_path = os.path.join(dirpath, dirname)
+        # Removes directories in place to prevent them and their children from
+        # being iterated. See https://docs.python.org/3/library/os.html#os.walk
+        if not self._ExcludeDir(full_dir_path):
+          filtered_dirnames.append(dirname)
+        # If a symlink is excluded above we don't want to print 2 messages.
```

Comment:
Nice! nit: maybe move the comment under the else block, makes it easier to associate.